cinema/views.py

```diff
@@ -34,6 +46,28 @@ class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.all()
     serializer_class = MovieSerializer
 
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        genres = self.request.query_params.get("genres")
+        actors = self.request.query_params.get("actors")
+        title = self.request.query_params.get("title")
+
+        if genres:
+            genres_ids = [int(str_id) for str_id in genres.split(",")]
+            queryset = Movie.objects.filter(genres__id__in=genres_ids)
+
+        if actors:
+            actors_ids = [int(str_id) for str_id in actors.split(",")]
+            queryset = queryset.filter(actors__id__in=actors_ids)
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+
+        if self.action == ("list", "retrieve"):
```

Comment:
The condition `self.action == ('list', 'retrieve')` is incorrect. The `self.action` attribute is a string representing the current action, such as 'list' or 'retrieve'. You should check each action separately, like `if self.action in ['list', 'retrieve']:`.